docs/howtos/exposing-rust-components-to-kotlin.md

```diff
@@ -0,0 +1,203 @@
+
+## How to expose your Rust Component to Kotlin
+
+Welcome!
+
+It's great
+that you've built a Rust Component
+and want to expose it to Koltin,
+but we haven't written a complete guide yet.
+
+Please share your :+1:
+on [the relevant github issue](https://github.com/mozilla/application-services/issues/599)
+to let us know
+that you wanted it.
+
+In the meantime,
+here are some preliminary notes:
+
+### High-level overview.
+
+The [fxa-client](/components/fxa-client) component
+provides a useful example.
+We assume you have written
+a nice core of Rust code
+in a [./src/](/components/fxa-client/src) directory
+for your component.
+
+First, you will need to
+flatten the Rust API
+into a set of C FFI bindings,
+conventially in a [./ffi/](/components/fxa-client/ffi) directory.
```

Comment:
FxA is one of the weirder ones, since it manually implements a lot more things than it needs to (this has perf benefits, and it works and is tested now, so I'm hesitant to remove it, but I also wouldn't recommend doing new things this way). 

I'd use logins or places as a better example.

Also it's maybe worth mentioning that some of the FFI bindings have to be exposed as traits from the nice core of rust code (you need to implement some traits for `ffi_support` to work, which must be done from within the crate where you defined them).

An example of this is https://github.com/mozilla/application-services/blob/master/components/logins/src/ffi.rs. Everything happens via macros (implement_into_ffi_by_json) except mapping rust errors to numeric error codes, which must be done manually.

This might be getting too into the weeds for a high level overview though